File: python/692_前K个高频单词_medium.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        cnt = collections.defaultdict(int)
        for word in words:
            cnt[word] += 1
        res = list()
        for word, val in cnt.items():
            res.append(word)
        def compare(a, b):
            if cnt[a] > cnt[b]:
                return -1
            elif cnt[a] == cnt[b]:
                if a < b:
                    return -1
                else:
                    return 1
            else:
                return 1
        res = sorted(res, key = functools.cmp_to_key(compare))
        return res[:k]

File: cpp/692_前K个高频单词_medium.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map <string, int> cnt;
        for (auto word : words)
            ++cnt[word];
        vector<string> res;
        for (auto [word, n] : cnt)
            res.push_back(word);
        sort(res.begin(), res.end(), [&](const string &a, const string &b){
            return cnt[a] == cnt[b] ? a < b : cnt[a] > cnt[b];
        });
        res.erase(res.begin()+k, res.end());
        return res;

    }
};